src/message_buffer.cpp

```diff
@@ -3,24 +3,90 @@
 #include <iomanip>
 #include <sstream>
 
-namespace quicr::messages {
+namespace quicr {
+uintVar_t
+to_varint(uint64_t v)
+{
+  assert(v < ((uint64_t)0x1 << 61));
+  return static_cast<uintVar_t>(v);
+}
+
+uint64_t
+from_varint(uintVar_t v)
+{
+  return static_cast<uint64_t>(v);
+}
+
+namespace messages {
+MessageBuffer::MessageBuffer(MessageBuffer&& other)
+  : _buffer{std::move(other._buffer)}
+{
+}
+
+MessageBuffer::MessageBuffer(const std::vector<uint8_t>& buffer)
+  : _buffer{buffer}
+{
+}
+
+MessageBuffer::MessageBuffer(std::vector<uint8_t>&& buffer)
+  : _buffer{std::move(buffer)}
+{
+}
+  
+void MessageBuffer::push_back(const std::vector<uint8_t>& data)
+{
+  _buffer.insert(_buffer.end(), data.begin(), data.end());
+}
+
+void MessageBuffer::pop_back(uint16_t len)
+{
+  const auto delta = _buffer.size() - len;
+  _buffer.erase(_buffer.begin() + delta, _buffer.end());
+};
+
+std::vector<uint8_t> MessageBuffer::back(uint16_t len)
+{
+  assert(len <= _buffer.size());
```

Comment:
While assert helps in debugging, we should constrain the release builds so they do not return more than the size() of octets.